src/components/SearchView.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { LibraryItem } from "./LibraryItem";
+import { LibraryView } from "../LibraryView";
+import { search, ItemData } from "../LibraryUtilities";
+
+interface SearchViewProps {
+    onSearchChanged: any;
+    libraryView: LibraryView;
+    items: ItemData[];
+}
+
+interface SearchViewStates {
+    searchText: string;
+}
+
+export class SearchView extends React.Component<SearchViewProps, SearchViewStates> {
+
+    constructor(props: SearchViewProps) {
+        super(props);
+
+        this.state = {
+            searchText: ''
+        };
+    }
+
+    performSearch(text: string) {
+        this.props.items.forEach(item => search(text, item));
+    }
+
+    generateStructuredItems() {
+        let structuredItems: JSX.Element[] = [];
+        let index = 0;
+        structuredItems = this.props.items.map((item: ItemData) => <LibraryItem key={index++} libraryView={this.props.libraryView} data={item} />);
+        return structuredItems;
+    }
+
+    setItemVisibility(visibility: boolean, items: ItemData[] = this.props.items) {
+        items.forEach(item => {
+            item.visible = visibility;
+            this.setItemVisibility(visibility, item.childItems);
+        });
+    }
```

Comment:
Move `setItemVisibility` to utility class, I don't think `this` is required at all.